### PR TITLE
Created - Create Server Screen

### PR DESCRIPTION
**Pull request by yugeshjain**

### CODE REVIEW

**Comment by Anmol92verma** [comment]
@yugeshjain requesting a screenshot in light/dark mode for merging this PR

**Comment by yugeshjain** [comment]
create_server_screen_light
create_server_screen_dark

**Comment by Anmol92verma** [comment]
@yugeshjain the icon tint in light theme, should be different

**Comment by yugeshjain** [comment]
It's an Image, its supposed to be colourful.. for now we just have a placeholder

File: app/src/main/java/dev/baseio/discordjetpackcompose/MainActivity.kt [hunk 1]
```diff
@@ -44,5 +52,10 @@ class MainActivity : ComponentActivity() {
                 }
             }
         }
+        WindowCompat.setDecorFitsSystemWindows(window, false)
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.R) {
+            window.addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN)
```
Comment (Anmol92verma) [inline app/src/main/java/dev/baseio/discordjetpackcompose/MainActivity.kt:57]: @yugeshjain can we add a comment here why this was needed ?

File: app/src/main/java/dev/baseio/discordjetpackcompose/ui/routes/dashboard/createServer/CreateServer.kt [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+            )
+        }
+    ) { paddingValues ->
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .background(create_server_screen),
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            CenteredTitleSubtitle(
```
Comment (Anmol92verma) [inline app/src/main/java/dev/baseio/discordjetpackcompose/ui/routes/dashboard/createServer/CreateServer.kt:69]: Perfect! reuse of common elements!

File: app/src/main/java/dev/baseio/discordjetpackcompose/ui/routes/onboarding/commonui/DiscordDialog.kt [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                            color = black
+                        ),
+                        modifier = Modifier.padding(
+                            top = 12.dp,
+                            start = 16.dp,
+                            end = 16.dp,
+                            bottom = 24.dp
+                        )
+                    )
+                    Divider(color = Color.LightGray, thickness = 0.5.dp)
```
Comment (Anmol92verma) [inline app/src/main/java/dev/baseio/discordjetpackcompose/ui/routes/onboarding/commonui/DiscordDialog.kt:86]: @yugeshjain saw this in another PR also, make sure we resolve the comments from there

File: app/src/main/java/dev/baseio/discordjetpackcompose/ui/routes/dashboard/createServer/CreateServerCard.kt [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+) {
+    Button(
+        colors = ButtonDefaults.buttonColors(backgroundColor = create_server_card_bg),
+        shape = RoundedCornerShape(8.dp),
+        modifier = Modifier
+            .padding(horizontal = 16.dp, vertical = 6.dp),
+        onClick = { onClick() },
+        contentPadding = PaddingValues(0.dp)
+    ) {
+        Row(
```
[the remark this answers is not in the document]
Comment (yugeshjain) [reply app/src/main/java/dev/baseio/discordjetpackcompose/ui/routes/dashboard/createServer/CreateServerCard.kt:45]: I used the clickable modifier but the onClick animation was not taking the Rounder Border, but button does that perfectly

File: app/src/main/java/dev/baseio/discordjetpackcompose/MainActivity.kt [hunk 5]
```diff
@@ -24,6 +29,9 @@ class MainActivity : ComponentActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+//        window.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE);
```
[the remark this answers is not in the document]
Comment (yugeshjain) [reply app/src/main/java/dev/baseio/discordjetpackcompose/MainActivity.kt:32]: Removed

File: app/src/main/java/dev/baseio/discordjetpackcompose/MainActivity.kt (hunk 1 above)
Comment (yugeshjain) [reply app/src/main/java/dev/baseio/discordjetpackcompose/MainActivity.kt:57]: Done

File: app/src/main/java/dev/baseio/discordjetpackcompose/ui/routes/dashboard/createServer/CreateServer.kt (hunk 2 above)
Comment (yugeshjain) [reply app/src/main/java/dev/baseio/discordjetpackcompose/ui/routes/dashboard/createServer/CreateServer.kt:69]: Thank You!!

File: app/src/main/java/dev/baseio/discordjetpackcompose/ui/routes/dashboard/createServer/CreateServerCard.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (yugeshjain) [reply app/src/main/java/dev/baseio/discordjetpackcompose/ui/routes/dashboard/createServer/CreateServerCard.kt:45]: ok, replaced it with surface, now its working

File: app/src/main/java/dev/baseio/discordjetpackcompose/ui/routes/dashboard/createServer/CreateServer.kt [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+                title = R.string.create_your_server,
+                subtitle = R.string.create_your_server_subtitle
+            )
+            Spacer(modifier = Modifier.size(8.dp))
+            LazyColumn(
+                modifier = Modifier
+                    .fillMaxHeight(0.8f)
+                    .background(create_server_screen)
```
[the remark this answers is not in the document]
Comment (yugeshjain) [reply app/src/main/java/dev/baseio/discordjetpackcompose/ui/routes/dashboard/createServer/CreateServer.kt:77]: Done.

**Pull request merged**